src/dbnode/x/xio/segment_reader.go

```diff
@@ -91,6 +94,8 @@ func (sr *segmentReader) Reset(segment ts.Segment) {
 func (sr *segmentReader) Finalize() {
 	// Finalize the segment
 	sr.segment.Finalize()
+	sr.lazyHead = nil
```

Comment:
This will have to be done in `Reset()` too, or you won't update `lazyHead` after resetting to a new segment

Reply:
Good call. Will fix

Reply:
Maybe have `sr.lazyHead = sr.lazyHead[:0]` and change the condition in `Read` changed to `b := sr.segment.Head; b != nil && len(sr.lazyHead) > 0` so it won't have to alloc more after Reset